lib/import.js

```diff
@@ -7,45 +11,84 @@ function txtImport(inputFilepath) {
     const text = fs.readFileSync(inputFilepath).toString();
     const partials = parseText(text);
     resolve(partials);
+  });
+}
+
+const smfImport = (inputPath) =>
+  new Promise((resolve) => {
```

Comment:
I thought I should keep consistency of all the functions of S2M.   
It may be difficult for users of this library to tell whether a function returns Promise or not.   
So I make all the function return promise.
I don't know what is the best. What do you think?

Reply:
hmm, if you wanto use Promise, you should use `fs.lstat` not `fs.lstatSync` since `fs.lstatSync` executes synchronously, so it's no reason to use Promise.

if you use "sync" function, you should just return `Promise.resolve`. 
you don't have to enclose with Promise.

Reply:
just `return Promise.resolve(smfs)`

Reply:
And I have one more question.  
How to tell if a function is synchronous or asynchronous?
I think s2m.txtImport function also has unnecessary promise wrapper.
But I can't tell if the parseText function which is iside the txtImport function is asynchronous or synchronous.

Reply:
if you want throw Error, `return Promise.reject(new Error("hoge"))`

Reply:
there're a lot of options.
1. write a API doc.
https://esdoc.org/

2. write an example
3. write d.ts
https://basarat.gitbooks.io/typescript/docs/types/ambient/d.ts.html

Reply:
Thanks. It is very good to introduce to this project. But now I'm thinking about whether the promise wrapper in declaration of textImport function is needed or not. 
```
function txtImport(inputFilepath) {
  return new Promise((resolve) => {
    console.log('Importing files...');
    const text = fs.readFileSync(inputFilepath).toString();
    const partials = parseText(text);
    resolve(partials);
  });
}
```
It uses subfunction, parseText. And paseText uses array.map and array.filter.  
In some case, I've seen Node go to the next function before the end of the evaluation in some cases such as array.forEach.
In the above example, It looks like working without Promise wrapper. But I'm not sure the following  functions wait until the end of the evaluation of all the array elements.